src/broadcast.jl

```diff
@@ -0,0 +1,29 @@
+Base.Broadcast._containertype(::Type{<:DataValue}) = DataValue
+
+Base.Broadcast.promote_containertype(::Type{Any}, ::Type{DataValue}) = DataValue
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Any}) = DataValue
+Base.Broadcast.promote_containertype(::Type{Tuple}, ::Type{DataValue}) = Tuple
+Base.Broadcast.promote_containertype(::Type{DataValue}, ::Type{Tuple}) = Tuple
```

Comment:
I'v added two more ``promote_containertype`` methods that handle the mixed ``DataValue`` and ``Nullable`` case, and say that it will return ``DataValue``. Did I do this correct? Was there another concern here that you had?